datameta/validation.py

```diff
@@ -135,11 +139,12 @@ def validate_submission_uniquekeys(db, db_files, db_msets):
         for db_mset in db_msets.values():
             for mdatrec in db_mset.metadatumrecords:
                 if mdatrec.metadatum.name == key:
-                    value_msets[mdatrec.value].append(db_mset)
+                    if mdatrec.value is not None:
```

Comment:
Could just be an `and` clause with the previous `if` condition, plain nested `if`s look somewhat funny IMHO